pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/HadoopPinotFS.java

```diff
@@ -121,7 +123,32 @@ public long length(URI fileUri) throws IOException {
 
   @Override
   public void copyToLocalFile(URI srcUri, File dstFile) throws Exception {
-    hadoopFS.copyToLocalFile(new Path(srcUri), new Path(dstFile.toURI()));
+    LOGGER.debug("starting to fetch segment from hdfs");
+    final String dstFilePath = dstFile.getAbsolutePath();
+    try {
+      final Path remoteFile = new Path(srcUri);
+      final Path localFile = new Path(dstFile.toURI());
+
+      RetryPolicy fixDelayRetryPolicy = RetryPolicies.fixedDelayRetryPolicy(retryCount, retryWaitMs);
+      fixDelayRetryPolicy.attempt(() -> {
+        try {
+          if (hadoopFS == null) {
+            throw new RuntimeException("hadoopFS client is not initialized when trying to copy files");
+          }
+          long startMs = System.currentTimeMillis();
+          hadoopFS.copyToLocalFile(remoteFile, localFile);
+          LOGGER.debug("copied {} from hdfs to {} in local for size {}, take {} ms", srcUri, dstFilePath,
+              dstFile.length(), System.currentTimeMillis() - startMs);
+          return true;
+        } catch (IOException ex) {
+          LOGGER.warn(String.format("failed to fetch segment %s from hdfs, might retry", srcUri), ex);
```

Comment:
It'd be good to just use LOGGER, i.e., replacing `%s` with `{}` and you can get rid of `String.format()`.